### PR TITLE
Encryption: Handle encryption key provider being a background service

### PR DESCRIPTION
**Pull request by undef1nd**
**What this PR does / why we need it**:
One of the encryption providers (enterprise part of the PR) needs to run a background job, so this PR introduces means to achieve that.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/secrets/manager/manager.go [hunk 1]
```diff
@@ -358,16 +360,30 @@ var (
 
 func (s *SecretsService) Run(ctx context.Context) error {
 	gc := time.NewTicker(gcInterval)
+	grp, gCtx := errgroup.WithContext(ctx)
+
+	for _, p := range s.providers {
+		if svc, ok := p.(registry.BackgroundService); ok {
```
Comment (xlson) [inline pkg/services/secrets/manager/manager.go:366]: I'd suggest adding a new interface that duplicated `registry.BackgroundService`, reusing the interface signals that this code is somehow related with wire and the registry when it actually isn't.

**Review by xlson — COMMENTED** [review]
Looks good! Just a minor suggested change.

File: pkg/services/secrets/manager/manager.go (hunk 1 above)
Comment (undef1nd) [reply pkg/services/secrets/manager/manager.go]: Thank you. Done :)

**Review by xlson — APPROVED** [review]
LGTM!

**Review by joanlopez — APPROVED** [review]
Looks great!!! 🚀

**Pull request merged**